### DISCUSSION TITLE
Does wails support wasm

### DISCUSSION THREAD

**Discussion by kcmvp** [wailsapp/wails#2707] [Q&A] [answered]
### Is your feature request related to a problem? Please describe.

May be it’s a very simple question, but I can’t find a sample app.

I am using wails build in-house tool, and I want to include a third party library which is a wasm. I want to know does wails support wasm and how to use it? 

### Describe the solution you'd like

Call wasm from wails

**Comment by leaanthony** [comment]
There is a Go library called Wazero that will interest you. You could bind a method and pass through the call to wazero, meaning you could call it from the frontend. I'm looking to build this into v3 as a plugin option

**Comment by kcmvp** [comment]
I suppose webview is already a wams runtime, why need another runtime as Wazero?

**Reply by kcmvp** [reply]
After a round search, found that webview does not support wasm either! If wails support wasm there will be infinite possibility, really hope it include this feature soon!

**Reply by leaanthony** [reply]
Watch this space 😉

**Reply by tassa-yoniso-manasi-karoto** [reply]
wasm works in webkit, at least right now.

when using with vite, for reference:
https://github.com/tassa-yoniso-manasi-karoto/langkit/blob/master/internal/gui/frontend/src/WASM_setup_considerations.md

**Comment by kcmvp** [comment]
To be more speicific, right now I am going to build vim editor base on wails & [vim.wams](https://github.com/rhysd/vim.wasm) or [vim.wams.react](https://github.com/rhysd/vim.wasm). This libray works well on desktop browser. 
just as you point out [wazero](https://github.com/tetratelabs/wazero) it's a wasm runtime. I suppose [wails](https://github.com/wailsapp/wails) + [wazero](https://github.com/tetratelabs/wazero) should behavior same as local **chrome**(Safair, Firefox).  is my understanding correct? 
search through the forum and find **wasm support** is on the **roadmap v3**.  I would really like to investigate this in advance, but I know less about wasm. can you give me some guide?  during the investigation I would give some useful feedback to the community(ask help 😉 )

**Reply by leaanthony** [reply]
No guarantees! Best to just try. If it's frontend WASM then you'll need to try on each webview per platform. Wkwebview on Mac, WebView2 on windows and webkit2gtk on linux

**Reply by kcmvp** [reply]
Thank you very much! you mean wails support front wams already but not test, right?

**Reply by leaanthony** [reply, marked as answer]
I'm saying No guarantees! Best to just try.

**Reply by kcmvp** [reply]
Thank you! I will have a try

**Comment by tataDan** [comment]
Will the Wazero plugin option be available in v3?

**Reply by leaanthony** [reply]
What's stopping you from using wazero now?
